src/modules/utils/motordrivercontrol/MotorDriverControl.cpp

```diff
@@ -461,14 +516,82 @@ void MotorDriverControl::set_options(Gcode *gcode)
+// 		('p' is typically for use with a no-op/status command to view the result of a previously written command.)
+//		('P' is typically for use with sending a status command and immediately sending it again to shift out, and print, its response.)
+// If 'xN' at end of hexdata, then repeats hexdata N times in buffer before writing -- typically used to write the same data to multiple similar devices that are daisy-chained on the SPI bus and similarly initialized/commanded.
+// Underscores '_' (and most other extraneous characters) between hexdata words denote the start of data for 'xN', otherwise ignored.
+// Comma separators or end of line will write previous data as an individual SPI write (SSEL) cycle.
+// Powerful but potentially hazardous, this writes free-form raw data and has no knowledge of what drivers are on the SPI bus, their word lengths, or how they are arranged on the bus.
+void MotorDriverControl::WriteReadSPIstr(const char* p, StreamOutput* stream)
 {
-    spi_cs_pin.set(0);
-    for (int i = 0; i < cnt; ++i) {
-        r[i]= spi->write(b[i]);
-    }
-    spi_cs_pin.set(1);
-    return cnt;
+	int x,xx;
+	uint8_t nib,paw=0,buf[40];
+	
+	//stream->printf("WriteReadSPIstr: %s\n",p);
+	
+	do {
+		buf[0]=0;
+		for (x=xx=0, nib=1; (x<40) && *p && (*p!=','); p++) {
+			if ((*p>='0') && (*p<='9'))		// Translate hex digits.
+				buf[x]=(buf[x]<<4)|(*p-'0');
+			else if ((*p>='A') && (*p<='F'))
+				buf[x]=(buf[x]<<4)|(*p-'A'+0xA);
+			else if (*p=='x') {		// Copy data n times in buffer.
+				int n=1;
+				if (!nib) buf[x++]<<=4, nib=1;  // This shouldn't happen, but in case last byte wasn't complete, do something consistent.
+				++p;
+				if ((*p>='2') && (*p<='9')) n=*p-'0';
+				while (--n) for (int i=x-xx; (x<40) && i; i--) buf[x++]=buf[xx++];
+				xx=x;
+				continue;
+			}
+			else if (((*p|0x20)=='p') && !x && nib) {  // 'p' or 'P' at beginning of line.
+				paw=*p;
+				continue;
+			}
+			else {	// Underscore '_' and other extraneous characters.
+				xx=x;
+				continue;
+			}
+			if (nib^=1) buf[++x]=0;
+		}
+		if (!nib) buf[x++]<<=4;  // This shouldn't happen, but in case last byte wasn't complete, do something consistent.
+		if (x) {
+			//if (stream) {
+			//	stream->printf("SPI <- ");
+			//	for (int n=0; n<x; n++) stream->printf("%02X",buf[n]);
+			//	stream->printf("\n");
+			//}
+			if (paw!='p') sendSPI(buf,x,NULL);  // Send (and dump received) data on SPI bus.
+			if (paw) {  // If print-after-write, then print received data.
+				sendSPI(buf,x,buf);	// Send and receive data on SPI bus.
+				if (stream) {
+					stream->printf("SPI -> ");
+					for (int n=0; n<x; n++) stream->printf("%02X",buf[n]);
+					stream->printf("\n");
+				}
+			}
+		}
+	} while (*p++==',');
 }
 
+
+// Called by the drivers codes to send and receive SPI data to/from the chip
+void MotorDriverControl::sendSPI(uint8_t *b, int cnt, uint8_t *r /* =0 */)
```

Comment:
There was no reason to change this to void.

Reply:
Yep, I got your first comment on this.  I'll change these back.  I was going to move it out to libs, but hadn't gotten around to it, yet.  It returns cnt, which is useless.  I was trying to clean some (simple) things up, but I get what you said about other PRs and potential conflicts.